### PR TITLE
add remapping surface area test

### PR DESCRIPTION
**Pull request by juliasloan25**
# PULL REQUEST

## Purpose and Content
Add a unit test to check that `LandSeaMask` remapping is conservative with respect to the surface area of the planet. Also change the remapping code to be monotone by default, removing the need for a `clean_mask` function.

## PR Checklist
- [x] This PR has a corresponding issue OR is linked to an SDI.
- [x] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [x] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [x] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [x] I linted my code on my local machine prior to submission OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code used in an integration test OR N/A.
- [x] All tests ran successfully on my local machine OR N/A.
- [x] All classes, modules, and function contain docstrings OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: experiments/AMIP/moist_mpi_earth/coupler_utils/regridder.jl [hunk 1]
```diff
@@ -67,7 +67,19 @@ function ncreader_rll_to_cgll_from_space(datafile_rll, varname, space; outfile =
         # write cgll mesh, overlap mesh and weight file 
         write_exodus(meshfile_cgll, topology)
         overlap_mesh(meshfile_overlap, meshfile_rll, meshfile_cgll)
-        remap_weights(weightfile, meshfile_rll, meshfile_cgll, meshfile_overlap; out_type = "cgll", out_np = Nq)
+
+        # in_np and mono arguments ensure mapping is conservative and monotone
```
Comment (LenkaNovak) [inline experiments/AMIP/moist_mpi_earth/coupler_utils/regridder.jl:71]: I would also comment here that np_in = dof = polynomial degree + 1

File: experiments/AMIP/moist_mpi_earth/coupler_utils/unit_tester.jl [hunk 2]
```diff
@@ -74,3 +80,52 @@ function test_interpol(boundary_space)
...
+
+    # init land-sea mask
+    land_mask = LandSeaMask(FT, mask_data, varname, boundary_space_t)
+
+    @test sum(land_mask) ≈ 4 * π * (radius^2)
+
+    rm(REGRID_DIR; recursive = true)
+end
+
+test_conservation_ncdataset()
```
Comment (LenkaNovak) [inline experiments/AMIP/moist_mpi_earth/coupler_utils/unit_tester.jl:131]: Do you mean `test_surfacearea_ones()` ?

**Review by LenkaNovak — CHANGES_REQUESTED** [review]
Good job, @juliasloan25 ! I left a couple of comments. Also, for the Buildkite runs, looks like you're using Nq 5 with the `mono` flag, which isn't enabled in TempestRemap yet. Can you change `--nh_poly` to 4, in the driver like you had before? Hopefully that'll fix it.

**Comment by juliasloan25** [comment]
bors try

**Comment by juliasloan25** [comment]
bors try

**Review by LenkaNovak — APPROVED** [review]

**Comment by LenkaNovak** [comment]
@juliasloan25 all is passing now and it's calling your CC commit for the `mono` flag. I will update the Manifest to the release version in a separate PR when we do other upgrades. After you've addressed the remaining issues above, feel free to squash and bors r+.

**Pull request merged**